### PR TITLE
Adds Deeply Read Test to DCR

### PR DESCRIPTION
**Pull request by nitro-marky**
## What does this change?
This sets up an AB test for the Deeply Read hypothesis, which replaces the second tab in the Most Popular container with articles that have achieved the longest read time/engagement.

Control:
Screenshot 2021-01-06 at 11 51 53

Variant:
Screenshot 2021-01-06 at 15 21 39

After a conversation with Brendan when tabs are clicked on an event is sent which should 🤞  be easy enough to query from the datalake. 

When clicking on a tab:

Screenshot 2021-01-06 at 15 17 21

When clicking on an article:
Screenshot 2021-01-06 at 15 17 48

Both contain Deeply Read so should be queryable.

### CODE REVIEW

File: src/web/components/MostViewed/MostViewedFooter/MostViewedFooter.tsx [hunk 1]
```diff
@@ -132,7 +137,12 @@ export const MostViewedFooter = ({
 				data-cy-ab-runnable-test={variantFromRunnable}
 			>
 				<section className={asideWidth}>
-					<h2 className={headingStyles}>Most popular</h2>
+					{!inDeeplyReadTestVariant && (
+						<h2 className={headingStyles}>Most popular</h2>
+					)}
+					{inDeeplyReadTestVariant && (
+						<h2 className={headingStyles}>Accross The Guardian</h2>
+					)}
```
Comment (oliverlloyd) [inline src/web/components/MostViewed/MostViewedFooter/MostViewedFooter.tsx:145]:
```suggestion
					{inDeeplyReadTestVariant ? (
						<h2 className={headingStyles}>Accross The Guardian</h2>
					) : (
						<h2 className={headingStyles}>Most popular</h2>
					)}
```

Comment (oliverlloyd) [reply src/web/components/MostViewed/MostViewedFooter/MostViewedFooter.tsx:145]: This is perhaps a subjective thing but I think this ternary would be nicer here?

File: src/web/components/MostViewed/MostViewedFooter/MostViewedFooterData.tsx [hunk 2]
```diff
@@ -50,13 +50,26 @@ function buildSectionUrl(
...
 	design,
 }: Props) => {
-	const url = buildSectionUrl(ajaxUrl, pillar, sectionName);
+	const ABTestAPI = useAB();
+
+	const inDeeplyReadTestVariant = ABTestAPI.isUserInVariant(
+		'DeeplyReadTest',
+		'variant',
+	);
+	// const url = buildSectionUrl(ajaxUrl, pillar, sectionName);
```
Comment (oliverlloyd) [inline src/web/components/MostViewed/MostViewedFooter/MostViewedFooterData.tsx:69]: Comment

File: src/web/components/MostViewed/MostViewedFooter/MostViewedFooterData.tsx [hunk 3]
```diff
@@ -50,13 +50,26 @@ function buildSectionUrl(
...
+	const url = inDeeplyReadTestVariant
+		? buildDeeplyReadUrl(ajaxUrl, pillar)
+		: buildSectionUrl(ajaxUrl, pillar, sectionName);
```
Comment (oliverlloyd) [inline src/web/components/MostViewed/MostViewedFooter/MostViewedFooterData.tsx:72]: This does read really nice so I'm hesitant if this works or not but ccould we maybe inline the joinUrl call? I guess it's a trade of of the abstraction meaning you have too move around the file to see what happening (plus you don't know if there's anything else using that code) vs. it being immediately clear what happening but not as pretty?

**Review by oliverlloyd — APPROVED** [review]
Really nice! I really like how this is super easy to review and understand!

Just a couple of take it or leave them comments

**Comment by oliverlloyd** [comment]
Note. The jest error `useAB must be used within the ABProvider` means there is a missing context provider. I _think_ this is a test issue

File: src/web/components/MostViewed/MostViewedFooter/MostViewedFooterData.tsx (hunk 3 above)
Comment (nitro-marky) [reply src/web/components/MostViewed/MostViewedFooter/MostViewedFooterData.tsx:72]: I guess my main reasoning for abstracting this way is that building a section URL is a bigger process and when/if it comes to removing the test logic it should be much easier to understand which parts relate to deeply read and which do not. Ultimately they both use joinUrl but I thought the difference between them merited having them split into their own functions.

File: src/web/components/MostViewed/MostViewedFooter/MostViewedFooter.tsx (hunk 1 above)
Comment (nitro-marky) [reply src/web/components/MostViewed/MostViewedFooter/MostViewedFooter.tsx:145]: 👍

File: src/web/components/MostViewed/MostViewedFooter/MostViewedFooterData.tsx (hunk 3 above)
Comment (oliverlloyd) [reply src/web/components/MostViewed/MostViewedFooter/MostViewedFooterData.tsx:72]: > when/if it comes to removing the test logic it should be much easier to understand which parts relate to deeply read and which do not

👍 that's a really sound reason - great point

**Comment by oliverlloyd** [comment]
I noticed we've added capitalisation to `Popular` but I think the intention is to have this as sentance case
Screenshot 2021-01-06 at 10 35 44

**Comment by nitro-marky** [comment]
> I noticed we've added capitalisation to `Popular` but I think the intention is to have this as sentance case
> Screenshot 2021-01-06 at 10 35 44

Do we want 'The' in 'Accross The Guardian to be lower case as well?

**Comment by oliverlloyd** [comment]
> Do we want 'The' in 'Accross The Guardian to be lower case as well?

My guess would be yes, for consistency - might be worth checking with @HarryFischer though

**Comment by oliverlloyd** [comment]
> > Do we want 'The' in 'Accross The Guardian to be lower case as well?
> 
> My guess would be yes, for consistency - might be worth checking with @HarryFischer though

That said. Actually, no, I think The Guardian is a noun so should be capitalised.

**Comment by nitro-marky** [comment]
> > > Do we want 'The' in 'Accross The Guardian to be lower case as well?
> > 
> > 
> > My guess would be yes, for consistency - might be worth checking with @HarryFischer though
> 
> That said. Actually, no, I think The Guardian is a noun so should be capitalised.

Yeah, you're right. I'll leave that then

**Comment by HarryFischer** [comment]
Yep to 'Most popular'
And it should be 'Across The Guardian'

**Pull request merged**